app/lib/katello/resources/candlepin/upstream_consumer.rb

```diff
@@ -9,6 +9,15 @@ def path(id = upstream_consumer_id)
             super(id)
           end
 
+          def get(organization)
```

Comment:
It looks like the UpstreamConsumer model already has a lot of this implemented. This approach seems to work just the same:

```
def get
  JSON.parse(resource.get)
end
```